src/Compilers/CSharp/Test/Semantic/Semantics/PatternSwitchTests.cs

```diff
@@ -1578,30 +1560,30 @@ static void M(object o)
                 // (49,22): error CS8129: No suitable 'Deconstruct' instance or extension method was found for type 'object', with 2 out parameters and a void return type.
                 //             if (o is (System.Int32, System.Int32)) {}
                 Diagnostic(ErrorCode.ERR_MissingDeconstruct, "(System.Int32, System.Int32)").WithArguments("object", "2").WithLocation(49, 22),
-                // (49,23): error CS0119: 'int' is a type, which is not valid in the given context
+                // (49,23): error CS8652: The feature 'type pattern' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //             if (o is (System.Int32, System.Int32)) {}
-                Diagnostic(ErrorCode.ERR_BadSKunknown, "System.Int32").WithArguments("int", "type").WithLocation(49, 23),
-                // (49,37): error CS0119: 'int' is a type, which is not valid in the given context
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "System.Int32").WithArguments("type pattern").WithLocation(49, 23),
+                // (49,37): error CS8652: The feature 'type pattern' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
                 //             if (o is (System.Int32, System.Int32)) {}
```

Comment:
>(System.Int32, System.Int32) [](start = 40, length = 28)

Hum, in the general case: `o is X x` and `o is X` mean related things, except that one also declares a variable. 

But for tuples, they mean different things:
- `o is (int, int) x` isn't a type test (it looks for a `Deconstruct`)
- `o is (int, int)` on the other hand is a type test for `System.ValueTuple<int, int>`.
Did I get that right?

This seems incongruous.

Reply:
No. `o is (int, int)` is not a type test.  It means the same thing but without declaring a variable. That would have been a syntax error in C# 8.